task/orchestrator_test.go

```diff
@@ -84,6 +92,30 @@ func TestOrchestrator(t *testing.T) {
 			gracePeriod: 2 * time.Second,
 			wantError:   "",
 		},
+		{
+			name:   "zombie processes are reaped",
```

Comment:
Added a test to ensure zombie process are indeed reaped on shutdown of the agent.